src/auth/auth.service.ts

```diff
@@ -83,6 +87,53 @@ export class AuthService {
     verifyData[email].expiredAt = null;
   }
 
+  async oauthMobileLogin(
+    data: OauthMobileLoginDto,
+  ): Promise<{ refreshToken: string; accessToken: string }> {
+    const oauthClient = new google.auth.OAuth2({
+      clientId: data.clientId,
+      clientSecret: oauthClientSecret,
+    });
+    const info = await oauthClient.getTokenInfo(data.idToken);
+    const email = info.email;
+
+    if (!info || !email) throw new NotFoundException('Not found oauth user');
+    else if (email.split('@')[1] !== 'gsm.hs.kr')
+      throw new ForbiddenException();
+
+    const replacedEmail = email.replace('@gsm.hs.kr', '');
+    const token = await this.getToken(replacedEmail);
+
+    if (
+      await this.userRepository.findOne({
+        where: { email: replacedEmail },
+      })
+    ) {
+      const hash = await bcrypt.hash(token.refreshToken, 10);
+
+      this.userRepository.update(replacedEmail, {
+        refreshToken: hash,
+      });
+
+      return token;
+    } else {
+      const register = this.jwtService.decode(data.idToken, { json: true });
```

Comment:
```suggestion
      const register = this.jwtService.decode(data.idToken);
```

`json: true`는 아마 default 값일 걸?